app/src/components/market/sections/market_sell/scalar_market_sell.tsx

```diff
@@ -0,0 +1,341 @@
+          `Could not compute amount of collateral to sell for '${amountShares.toString()}' and '${holdingsOfSoldOutcome.toString()}'`,
+        )
+        return [null, null, null, null]
+      }
+
+      const balanceAfterTrade = computeBalanceAfterTrade(
+        balances.map(b => b.holdings),
+        positionIndex,
+        amountToSell.mul(-1),
+        amountShares.mul(-1),
+      )
+
+      const pricesAfterTrade = MarketMakerService.getActualPrice(balanceAfterTrade)
+      const potentialValue = mulBN(amountToSell, 1 / (1 - marketFeeWithTwoDecimals))
+      const costFee = potentialValue.sub(amountToSell)
+
+      const newPrediction = pricesAfterTrade[1] * ((upperBound || 0) - (lowerBound || 0)) + (lowerBound || 0)
+
+      logger.log(`Amount to sell ${amountToSell}`)
+      return [costFee, newPrediction, amountToSell, potentialValue]
+    },
+    [balances, positionIndex, lowerBound, upperBound, fee],
+  )
+
+  const [costFee, newPrediction, tradedCollateral, potentialValue] = useAsyncDerivedValue(
+    amountShares || Zero,
+    [new BigNumber(0), 0, amountShares, new BigNumber(0)],
+    calcSellAmount,
+  )
+
+  const formattedNewPrediction =
+    newPrediction && (newPrediction - (lowerBound || 0)) / ((upperBound || 0) - (lowerBound || 0))
+
+  const finish = async () => {
+    const outcomeIndex = positionIndex
+    try {
+      if (!tradedCollateral) {
+        return
+      }
+
+      const sharesAmount = formatBigNumber(amountShares || Zero, collateral.decimals)
+
+      setStatus(Status.Loading)
+      setMessage(`Selling ${sharesAmount} shares ...`)
+
+      const cpk = await CPKService.create(provider)
+
+      await cpk.sellOutcomes({
+        amount: tradedCollateral,
+        conditionalTokens,
+        marketMaker,
+        outcomeIndex,
+      })
+
+      await fetchGraphMarketMakerData()
+
+      setAmountShares(null)
+      setAmountSharesToDisplay('')
+      setStatus(Status.Ready)
+      setMessage(`Successfully sold ${sharesAmount} '${balances[outcomeIndex].outcomeName}' shares.`)
+    } catch (err) {
+      setStatus(Status.Error)
+      setMessage(`Error trying to sell '${balances[outcomeIndex].outcomeName}' Shares.`)
+      logger.error(`${message} - ${err.message}`)
+    }
+    setIsModalTransactionResultOpen(true)
+  }
+
+  const selectedOutcomeBalance = formatNumber(formatBigNumber(balanceItem.shares, collateral.decimals))
+
+  const amountError =
+    balanceItem.shares === null
+      ? null
+      : balanceItem.shares.isZero() && amountShares?.gt(balanceItem.shares)
+      ? `Insufficient balance`
+      : amountShares?.gt(balanceItem.shares)
+      ? `Value must be less than or equal to ${selectedOutcomeBalance} shares`
+      : null
+
+  const isSellButtonDisabled =
+    !amountShares ||
+    (status !== Status.Ready && status !== Status.Error) ||
+    amountShares?.isZero() ||
+    amountError !== null ||
+    isNegativeAmountShares
+
+  const dust = parseUnits('0.00001', collateral.decimals)
```

Comment:
The `isDust` function from https://github.com/protofire/omen-exchange/pull/1411 should be incorporated here, along with all its implementations. This will fix: https://github.com/protofire/omen-exchange/issues/1485